### PR TITLE
Add task list and activity panel E2E tests

### PR DESCRIPTION
**Pull request by louwie17**
Added E2E tests for hiding and showing the task list, and some simple tests for the activity panels.
I made use of `@woocommerce/api` and `@woocommerce/e2e-utils` for creating products and orders on the fly.

No changelog as it is in the package.

### Detailed test instructions:

- Run E2E tests locally `npm install && npx wc-e2e docker:up` and then `npx wc-e2e test:e2e`
- Check if the github action passed correctly

### CODE REVIEW

File: tests/e2e/docker/initialize.sh [hunk 1]
```diff
@@ -11,3 +11,6 @@ wp plugin install woocommerce --activate
 wp theme install twentynineteen --activate
 wp plugin activate woocommerce-admin
 wp user create customer customer@woocommercecoree2etestsuite.com --user_pass=password --role=customer --path=/var/www/html
+
+# we cannot create API keys for the API, so we using basic auth, this plugin allows that.
+wp plugin install https://github.com/WP-API/Basic-Auth/archive/master.zip --activate
```
Comment (louwie17) [inline tests/e2e/docker/initialize.sh:16]: Had to add this to make sure the `@woocommerce/api` works correctly.

File: packages/admin-e2e-tests/src/specs/homescreen/activity-panel.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		beforeAll( async () => {
+			await login.login();
+
+			// This makes this test more isolated, by always navigating to the
+			// profile wizard and skipping, this behaves the same as if the
+			// profile wizard had not been run yet and the user is redirected
+			// to it when trying to go to wc-admin.
+			await withRestApi.deleteAllProducts();
+			await removeAllOrders();
+			await updateOption( 'woocommerce_task_list_hidden', 'no' );
```
Comment (louwie17) [inline packages/admin-e2e-tests/src/specs/homescreen/activity-panel.ts:36]: I might have to put this in a helper function, given we might use this more often.

**Review by ilyasfoo — APPROVED** [review]
This looks great, @louwie17! Sorry it had to wait for so long to get reviewed. I only have a couple of minor comments.

I should note that one of the test may be abit brittle, on some runs I've got this:

```
(node:32985) UnhandledPromiseRejectionWarning: Error: Protocol error (Runtime.callFunctionOn): Target closed.
    at [redacted]/woocommerce-admin/node_modules/puppeteer/lib/Connection.js:183:56
    at new Promise (<anonymous>)
    at CDPSession.send ([redacted]/woocommerce-admin/node_modules/puppeteer/lib/Connection.js:182:12)
    at ExecutionContext._evaluateInternal ([redacted]/woocommerce-admin/node_modules/puppeteer/lib/ExecutionContext.js:107:44)
    at ExecutionContext.evaluateHandle ([redacted]/woocommerce-admin/node_modules/puppeteer/lib/ExecutionContext.js:57:17)
    at ExecutionContext.<anonymous> ([redacted]/woocommerce-admin/node_modules/puppeteer/lib/helper.js:112:23)
    at WaitTask.rerun ([redacted]/woocommerce-admin/node_modules/puppeteer/lib/DOMWorld.js:570:65)
  -- ASYNC --
    at Frame.<anonymous> ([redacted]/woocommerce-admin/node_modules/puppeteer/lib/helper.js:111:15)
    at Page.waitForSelector ([redacted]/woocommerce-admin/node_modules/puppeteer/lib/Page.js:1122:29)
    at WcHomescreen.next ([redacted]/woocommerce-admin/packages/admin-e2e-tests/build/pages/WcHomescreen.js:100:47)
    at Generator.next (<anonymous>)
    at [redacted]/woocommerce-admin/packages/admin-e2e-tests/build/pages/WcHomescreen.js:8:71
    at new Promise (<anonymous>)
    at __awaiter ([redacted]/woocommerce-admin/packages/admin-e2e-tests/build/pages/WcHomescreen.js:4:12)
    at WcHomescreen.getActivityPanels ([redacted]/woocommerce-admin/packages/admin-e2e-tests/build/pages/WcHomescreen.js:99:16)
    at WcHomescreen.<anonymous> ([redacted]/woocommerce-admin/packages/admin-e2e-tests/build/pages/WcHomescreen.js:119:47)
    at Generator.next (<anonymous>)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:32985) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
(node:32985) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
 FAIL  ../../../tests/e2e/specs/homescreen/activity-panel.test.ts (42.009s)
  Homescreen activity panel
    ✓ should not show activity panel while task list is displayed (473ms)
    ✓ should not show panels when there are no orders or products yet with task list hidden (7127ms)
    ✓ should show Reviews panel when we have at-least one product (3128ms)
    ✓ should show Orders and Stock panels when at-least one order is added (2830ms)
    Orders panel
      ✓ should show: "you have fullfilled all your orders" when expanding Orders panel if no actionable orders (872ms)
      ✕ should show actionable Orders when expanding Orders panel (794ms)

  ● Homescreen activity panel › Orders panel › should show actionable Orders when expanding Orders panel

    Element h4 (text: "You’ve fulfilled all your orders") not found

      JSHandles can be evaluated only in the context they were created!

```

Is it possible that the activity panel may have animations? Though I should say this shouldn't be a blocker to get this merged since it's quite difficult to reproduce. Pre-approving!

File: packages/admin-e2e-tests/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+# Unreleased
+
+-   Add test cases for home screent tasklist and activity panels. #7509
```
Comment (tomalec) [inline packages/admin-e2e-tests/CHANGELOG.md:3]: Looks like a typo in "screen**t**", Grammarly result:
```suggestion
-   Add test cases for the home screen tasklist and activity panels. #7509
```

File: packages/admin-e2e-tests/src/specs/homescreen/activity-panel.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+		afterAll( async () => {
+			await withRestApi.deleteAllProducts();
+			await removeAllOrders();
+			await updateOption( 'woocommerce_task_list_hidden', 'no' );
+			await login.logout();
+		} );
+
+		it( 'should not show activity panel while task list is displayed', async () => {
+			expect( await homeScreen.isTaskListDisplayed() ).toBe( true );
```
Comment (tomalec) [inline packages/admin-e2e-tests/src/specs/homescreen/activity-panel.ts:52]: :nail_care:  Just a cosmetic idea.
WDYT of replacing all `expect( await … )` with `await expect( … ).resolves`?
That should give, nicer assertion failure messages, when promise rejects. (https://jestjs.io/docs/expect#resolves)
```suggestion
			await expect( homeScreen.isTaskListDisplayed() ).resolves.toBe( true );
```

File: packages/admin-e2e-tests/src/specs/homescreen/activity-panel.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+			expect( await homeScreen.isActivityPanelShown() ).toBe( false );
+		} );
+
+		it( 'should show Reviews panel when we have at-least one product', async () => {
+			await createSimpleProduct( simpleProductName, '9.99' );
+			await page.reload( {
+				waitUntil: [ 'networkidle0', 'domcontentloaded' ],
+			} );
+			const activityPanels = await homeScreen.getActivityPanels();
+			expect( activityPanels.length ).toBe( 1 );
```
Comment (tomalec) [inline packages/admin-e2e-tests/src/specs/homescreen/activity-panel.ts:68]: More on nicer assertion messages?
```suggestion
			expect( activityPanels ).toHaveLength( 1 );
```

File: packages/admin-e2e-tests/src/specs/homescreen/activity-panel.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+			expect(
+				activityPanels.findIndex( ( p ) => p.title === 'Reviews' )
+			).toBeGreaterThanOrEqual( 0 );
```
Comment (tomalec) [inline packages/admin-e2e-tests/src/specs/homescreen/activity-panel.ts:71]:
```suggestion
			expect( activityPanels ).toContain(
				expect.objectContaining( { title: 'Reviews' } )
			);
```
To get something more descriptive than "Expected -1 to be greater than on equal 0"

File: packages/admin-e2e-tests/src/specs/homescreen/activity-panel.ts (hunk 4 above)
Comment (louwie17) [reply packages/admin-e2e-tests/src/specs/homescreen/activity-panel.ts:52]: Oooh nice, I didn't know that was an option, this does look nicer.

File: packages/admin-e2e-tests/src/specs/homescreen/activity-panel.ts (hunk 6 above)
Comment (louwie17) [reply packages/admin-e2e-tests/src/specs/homescreen/activity-panel.ts:71]: Nice another one that I was not are off 👍

**Comment by louwie17** [comment]
@ilyasfoo @tomalec I addressed your changes, would either one of you give this a quick once over again? 😄

**Comment by ilyasfoo** [comment]
@louwie17 Your changes look great! 🚢

**Pull request merged**